### ISSUE TITLE
ORDER BY does not handle composite expressions

### ISSUE THREAD

**Issue by PauloAmora** [cmu-db/peloton#685]
Given the following scenario:

```SQL

CREATE TABLE foo (a INT PRIMARY KEY, b INT, c INT);
INSERT INTO foo VALUES (1, 22, 333);
INSERT INTO foo VALUES (2, 11, 444);
INSERT INTO foo VALUES (3, 00, 555);
```
The command `SELECT * FROM foo ORDER BY a+c` crashes the database.

**Comment by haojin2** [comment]
My guess would be that we hit exception in OrderByTransform. Seems to me that "a+c" would be an AExpr parse node but we only try transforming Expr parse nodes in the function.

**Comment by PauloAmora** [comment]
Yes, I just checked that on `SimpleOptimizer`, the sort columns are not added to the plan. I'm not sure how this is handled in the new optimizer, but the tests pass locally and fail on travis.

**Comment by haojin2** [comment]
This should be handled in postgresparser. We need to add a call to AExprTransform in OrderByTransform function.

**Comment by apavlo** [comment]
We are going to ditch the SimpleOptimizer after we merge #640

**Comment by PauloAmora** [comment]
On the tests this is covered (`optimizer_sql_tests`) and looks ok. Do I close this?

**Comment by apavlo** [comment]
Yes, I think we support this now. Is this correct @hzxa21 @chenboy @luochenUmich ?

**Comment by chenboy** [comment]
Yes, the new optimizer supports expressions.

**Comment by apavlo** [comment]
Fixed in #640
